src/index.js

```diff
@@ -80,23 +80,27 @@
         inputErrorBorder = '2px solid red';
 
     var FormValidator = function (form, settings) {
-        if (!form || (settings && typeof settings !== 'object')) {
-            return;
-        }
-
         if (!(this instanceof FormValidator)) {
             return new FormValidator(form, settings);
         }
 
+        /**
+         * The settings object argument is optional and if is defined
+         * and it wasn't an object then an empty object returned.
+         */
+        if (!form || settings === null || (typeof settings !== 'undefined' && typeof settings !== 'object')) {
+            return {};
+        }
+
         this.form = form instanceof Element ? form : document.querySelector(form);
         this.events = (settings && settings.events && this.initEvents(settings.events)) || defaultEvents;
-        this.showErrors = typeof (settings && settings.showErrors) === 'undefined' ? showErrors : settings.showErrors;
+        this.showErrors = typeof (settings && settings.showErrors) !== 'undefined' ? settings.showErrors : showErrors;
```

Comment:
When you reach this point of your code, it means settings already got checked [line 91](https://github.com/AmraniCh/form-validate-js/blob/8d37024005d6712ff4a018127a4c81f9b40b8605/src/index.js#L91) and it is going to be either the object that the user passed or the default value '{}'.
So it is redundant to do the following :
```js
typeof (settings && settings.showErrors) !== 'undefined'
```
It can be replaced with :
```js
settings.showErrors || showErrors
```